src/software/util/parameter/generate_parameters_v2.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/env python
+"""
+This script runs as a gen_rule for bazel build //software/util/parameter and
+generates the necessary c++ files before compiling to setup DynamicParameters
+
+"""
+import yaml
+import os
+import time
+import sys
+import constants
+
+#######################################################################
+#                              Load Yaml                              #
+#######################################################################
+
+
+def load_configuration(paths_to_yaml: list):
+    """Loads the yaml files in the current directory and
+    makes a dictionary containing the parameter name and its
+    keys with the proper hierarchy
+
+    NOTE: No exception handling in place so that when an error happens
+    it raises to the main thread when bazel build runs
+
+    :param paths_to_yaml: The paths to where the YAML files are stored
+    :type paths_to_yaml: list of str
+    :returns: Dictionary containing the parameters with proper hierarchy
+    :rtype: dict
+
+    """
+    param_info = {}
+
+    # parse yaml and create parameter dictionaries
+    for filename in paths_to_yaml:
+        with open(filename, 'r') as param_yaml:
+            try:
+                config_descriptions = yaml.load(param_yaml)
+                param_info.update(config_descriptions)
+            except yaml.YAMLError as ymle:
+                error_msg = "{} could not be loaded correctly, please check format".format(
+                    filename)
+                print(constants.AUTOGEN_FAILURE_MSG.format(error_msg))
+                sys.exit(error_msg)
+    return param_info
+
+#######################################################################
+#                              Parameter                              #
+#######################################################################
+
+
+class Parameter(object):
+
+    """Container to hold the required data to generate
+    a Parameter that can be in any Config
+
+    :param parameter_name: the name of the parameter
+    :param parameter_description: the yaml configuration for the parameter 
+    :type parameter_name: str
+    :type parameter_description: dict
+
+    """
+
+    def __init__(self, parameter_name: str, parameter_description: dict):
+
+        """INITIALIZER"""
```

Comment:
I'm not sure this comment adds much value.

Reply:
yeah my snipped popped it here didn't notice it, its gone now